lib/modelMaterialsCommon.js

```diff
@@ -727,6 +729,85 @@ function lightDefaults(gltf) {
     }
 }
 
+function getShaderVariable(accessor) {
+  var type = accessor.type;
+  if (type === 'SCALAR') {
+    return 'float';
+  }
+  return type.toLowerCase();
+}
+
+function ensureSemanticExistenceForPrimitive(gltf, primitive) {
+  var accessors = gltf.accessors;
+  var materials = gltf.materials;
+  var techniques = gltf.techniques;
+  var programs = gltf.programs;
+  var shaders = gltf.shaders;
+
+  var attributes = primitive.attributes;
+  var materialId = primitive.material;
+  var material = materials[materialId];
+  var techniqueId;
+  var technique;
+  if (defined(material)) {
+    techniqueId = material.technique;
+    technique = techniques[techniqueId];
+  }
+  var programId;
+  var program;
+  if (defined(technique)) {
+    programId = technique.program;
+    program = programs[programId];
+  }
+  var vertexShaderId;
+  var vertexShader;
+  if (defined(program)) {
+    vertexShaderId = program.vertexShader;
+    vertexShader = shaders[vertexShaderId];
+  }
+
+  for (var semantic in attributes) {
+    if (attributes.hasOwnProperty(semantic)) {
+      if (!defined(techniqueParameterForSemantic(technique, semantic))) {
+        if (defined(technique)) {
+          var accessorId = attributes[semantic];
+          var accessor = accessors[accessorId];
+          var lowerCase = semantic.toLowerCase();
+          var attributeName = 'a_' + lowerCase;
+          technique.parameters[lowerCase] = {
+            semantic: semantic,
+            type: accessor.componentType
+          };
+          technique.attributes[attributeName] = lowerCase;
+          if (defined(program)) {
+            program.attributes.push(attributeName);
+            if (defined(vertexShader)) {
+              var pipelineExtras = vertexShader.extras._pipeline;
+              var shaderText = pipelineExtras.source.toString();
+              shaderText = 'attribute ' + getShaderVariable(accessor) + ' ' + attributeName + ';\n' + shaderText;
+              pipelineExtras.source = shaderText.toString();
+            }
+          }
+        }
+      }
+    }
+  }
+}
+
+function ensureSemanticExistence(gltf) {
+  var meshes = gltf.meshes;
+  for (var meshId in meshes) {
+    if (meshes.hasOwnProperty(meshId)) {
+      var mesh = meshes[meshId];
+      var primitives = mesh.primitives;
+      var primitivesLength = primitives.length;
+      for (var i = 0; i < primitivesLength; i++) {
+        ensureSemanticExistenceForPrimitive(gltf, primitives[i]);
+      }
+    }
+  }
+}
```

Comment:
If the function above be can be shorted enough, I wouldn't see a problem combining the two functions.